### ISSUE TITLE
Integrate AST exporter JSON plugin in the official clang repository?

### ISSUE THREAD

**Issue by aryx** [facebook/facebook-clang-plugins#24]
Have you considered adding the ASTExporter.cpp in the clang repository itself?
Right now it is pretty tedious to install facebook-clang-plugins because it requires to recompile your own clang,
and to install it, and to override existing one. Wouldn't it be simpler if the plugin was already in clang
so we could just call the regular clang binary?

**Comment by jvillard** [comment]
The AST Exporter is only used by Infer, and so this entire repo has been merged into the infer repo: https://github.com/facebook/infer/ and development continues there. We are planning on archiving this repo in the near future.

Because the AST Exporter is only used for infer, I don't think it would have high chances of getting into the clang repositories, but I could be wrong about that. Do you have another use for this plugin?

**Comment by aryx** [comment]
Yes I would love to leverage the clang AST from OCaml in semgrep: https://semgrep.dev/

See also https://github.com/returntocorp/semgrep

**Comment by aryx** [comment]
More generally any OCaml programmer who would like to to stuff on c++ would benefit from an easy-to-install OPAM library that can leverage clang C++ ASTs. Just like what CIL did for C 20 years ago
https://github.com/cil-project/cil

**Comment by jvillard** [comment]
Not saying you shouldn't use this plugin (we use it in production and it works well), but have you looked at clangml? https://gitlab.inria.fr/tmartine/clangml

**Comment by aryx** [comment]
Oh no I didn't know about clangml! Thx! I'll have a look.
